api-tests/src/utils/jar-generator.ts

```diff
@@ -0,0 +1,43 @@
+import * as jose from "jose";
+import config from "../config/config.js";
+import fs from "node:fs";
+import * as path from "path";
+import { fileURLToPath } from "url";
+import { getRandomString } from "./random-string-generator.js";
+import { createSignedJwt } from "./jwt-signer.js";
+
+const encAlg = "RSA-OAEP-256";
+const encMethod = "A256GCM";
+const encKey = await jose.importJWK(
+  JSON.parse(config.CORE_BACK_PUBLIC_ENCRYPTION_KEY) as jose.JWK,
+  encAlg,
+);
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+export const generateJar = async (
+  subject: string,
+  journeyType: string,
+): Promise<string> => {
+  const payloadData = JSON.parse(
+    fs.readFileSync(
```

Comment:
Yep can do - but in this scenario why does it matter? We're going to just wait for it anyway?

Reply:
It's a good habit because there is pretty much never a good reason to use synchronous I/O in node.

Although you might think `await asyncCall()` is equivalent to doing it synchronously (and in Java it almost is), in Node this will block the entire event loop - meaning no other code can execute while waiting for it, whereas using async/await does not. In a webserver this is pretty fatal (all requests are blocked!), but even here, we have no idea what other processes cucumber might be trying to run in the background.

*(aside from some _implementation details_)

Reply:
Ah, interesting. Thanks